src/PathUtilities.cpp

```diff
@@ -0,0 +1,302 @@
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+
+#include "PathUtilities.h"
+
+namespace Mezzanine {
+namespace Filesystem {
+    ///////////////////////////////////////////////////////////////////////////////
+    // Path Utilities
+
+    String GetDirName(const char* FileName)
+    {
+        size_t SlashPos = String::npos, CurrPos = 0;
+        while( FileName[CurrPos] != '\0' )
+        {
+            if( FileName[CurrPos] == '\\' || FileName[CurrPos] == '/' ) {
+                SlashPos = CurrPos;
+            }
+            ++CurrPos;
+        }
+        if( SlashPos != String::npos ) {
+            return String(FileName,SlashPos + 1);
+        }
+        return String();
+    }
+
+    String GetDirName(const StringView FileName)
+    {
+        size_t SlashPos = FileName.find_last_of("\\/");
+        if( FileName.npos == SlashPos ) {
+            return String();
+        }else{
+            return String( FileName.substr(0,SlashPos + 1) );
+        }
+    }
+
+    String GetBaseName(const char* FileName)
+    {
+        size_t SlashPos = String::npos, CurrPos = 0;
+        while( FileName[CurrPos] != '\0' )
+        {
+            if( FileName[CurrPos] == '\\' || FileName[CurrPos] == '/' ) {
+                SlashPos = CurrPos;
+            }
+            ++CurrPos;
+        }
+        if( SlashPos != String::npos ) {
+            return String(FileName + (SlashPos + 1),(CurrPos - SlashPos) - 1);
+        }
+        return String(FileName,CurrPos);
+    }
+
+    String GetBaseName(const StringView FileName)
+    {
+        size_t SlashPos = FileName.find_last_of("\\/");
+        if( FileName.npos == SlashPos ) {
+            return String( FileName );
+        }else{
+            return String( FileName.substr(SlashPos + 1,FileName.npos) );
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Absolute and Relative Paths
+
+    Boole IsPathAbsolute(const StringView ToCheck)
+    {
+        return IsPathAbsolute_Windows(ToCheck) || IsPathAbsolute_Posix(ToCheck);
+    }
+
+    Boole IsPathAbsolute_Host(const StringView ToCheck)
+    {
+    #ifdef MEZZ_WINDOWS
+        return IsPathAbsolute_Windows(ToCheck);
+    #else
+        return IsPathAbsolute_Posix(ToCheck);
+    #endif
+    }
+
+    Boole IsPathAbsolute_Posix(const StringView ToCheck)
+    {
+        if( !ToCheck.empty() ) {
+            return ( ToCheck[0] == '/' && ToCheck.find_first_of('\\') == String::npos );
```

Comment:
this can be shortened to `return ToCheck[0] == '/';` unless we want to enforce opinions here.

Reply:
Shortened.  I was mistakenly under the impression that '/' at the start could be used on windows as a valid relative path.  So I wanted to add an extra check for as much sanity as I could get.  Testing on cmd demonstrated that wasn't the case.